nano/core_test/active_transactions.cpp

```diff
@@ -11,60 +11,62 @@ using namespace std::chrono_literals;
-	node_flags2.disable_rep_crawler = true;
-	auto & node2 = *system.add_node (node_config2, node_flags2);
+	nano::system system{};
 	system.deadline_set (5s);
-	// Let node2 know about the block
-	while (node2.active.empty ())
-	{
-		node1.network.flood_block (send, nano::buffer_drop_policy::no_limiter_drop);
-		ASSERT_NO_ERROR (system.poll ());
-	}
-	// Save election to check request count afterwards
-	auto election = node2.active.election (send->qualified_root ());
-	ASSERT_NE (nullptr, election);
+
+	nano::node_flags node_flags1{};
+	node_flags1.disable_request_loop = true;
+	auto & node1 = *system.add_node (node_flags1);
+
+	nano::node_config node_config2{ nano::get_available_port (), system.logging };
+	node_config2.frontiers_confirmation = nano::frontiers_confirmation_mode::disabled;
+	auto & node2 = *system.add_node (node_config2);
+
+	auto send1 = nano::send_block_builder ()
+				 .previous (nano::dev::genesis->hash ())
+				 .destination (nano::public_key ())
+				 .balance (nano::dev::constants.genesis_amount - 100)
+				 .sign (nano::dev::genesis_key.prv, nano::dev::genesis_key.pub)
+				 .work (*system.work.generate (nano::dev::genesis->hash ()))
+				 .build_shared ();
+
+	node1.process_active (send1);
+	ASSERT_TIMELY (5s, node1.active.election (send1->qualified_root ()) != nullptr);
+
+	std::shared_ptr<nano::election> election{};
+	ASSERT_TIMELY (5s, (election = node2.active.election (send1->qualified_root ())) != nullptr);
+
 	// Add key to node1
 	system.wallet (0)->insert_adhoc (nano::dev::genesis_key.prv);
-	// Add representative to disabled rep crawler
-	auto peers (node2.network.random_set (1));
-	ASSERT_FALSE (peers.empty ());
-	{
-		nano::lock_guard<nano::mutex> guard (node2.rep_crawler.probable_reps_mutex);
-		node2.rep_crawler.probable_reps.emplace (nano::dev::genesis_key.pub, nano::dev::constants.genesis_amount, *peers.begin ());
-	}
-	ASSERT_TIMELY (5s, election->votes ().size () != 1); // Votes were inserted (except for the null account)
-	auto confirm_req_count (election->confirmation_request_count.load ());
-	// At least one confirmation request
-	ASSERT_GT (confirm_req_count, 0u);
-	ASSERT_FALSE (election->confirmed ()); // Cannot be confirmed without final vote
-	node1.confirmation_height_processor.add (send); // Confirm block for node1 for final vote generation
-	ASSERT_TIMELY (5s, node1.ledger.block_confirmed (node1.store.tx_begin_read (), send->hash ()));
-	node1.history.erase (send->root ()); // Have to erase existing non-final vote for final vote generation (at runtime it should be replaced with automatically generated final vote from election)
+
+	// Vote was inserted (+ not_an_account)
+	ASSERT_TIMELY (5s, election->votes ().size () == 2);
+	// Cannot be confirmed without final vote
+	ASSERT_FALSE (election->confirmed ());
+
+	// TODO: investigate
+	//
+	// auto const confirm_req_count = election->confirmation_request_count.load();
```

Comment:
@clemahieu -- need a bit of help with this one. It fails randomly, but definitely reproducible. I'd actually say it fails more than it passes. Any thoughts why the confirmation requests wouldn't be seen?